containerd-shim/process.go

```diff
@@ -115,185 +113,21 @@ func loadCheckpoint(checkpointPath string) (*checkpoint, error) {
 	return &cpt, nil
 }
 
-func (p *process) start() error {
-	cwd, err := os.Getwd()
-	if err != nil {
-		return err
-	}
-	logPath := filepath.Join(cwd, "log.json")
-	args := append([]string{
-		"--log", logPath,
-		"--log-format", "json",
-	}, p.state.RuntimeArgs...)
-	if p.state.Exec {
-		args = append(args, "exec",
-			"--process", filepath.Join(cwd, "process.json"),
-			"--console", p.consolePath,
-		)
-	} else if p.checkpoint != nil {
-		args = append(args, "restore",
-			"--image-path", p.checkpointPath,
-			"--work-path", filepath.Join(p.checkpointPath, "criu.work", "restore-"+time.Now().Format(time.RFC3339)),
-		)
-		add := func(flags ...string) {
-			args = append(args, flags...)
-		}
-		if p.checkpoint.Shell {
-			add("--shell-job")
-		}
-		if p.checkpoint.TCP {
-			add("--tcp-established")
-		}
-		if p.checkpoint.UnixSockets {
-			add("--ext-unix-sk")
-		}
-		if p.state.NoPivotRoot {
-			add("--no-pivot")
-		}
-	} else {
-		args = append(args, "start",
-			"--bundle", p.bundle,
-			"--console", p.consolePath,
-		)
-		if p.state.NoPivotRoot {
-			args = append(args, "--no-pivot")
-		}
-	}
-	args = append(args,
-		"-d",
-		"--pid-file", filepath.Join(cwd, "pid"),
-		p.id,
-	)
-	cmd := exec.Command(p.runtime, args...)
-	cmd.Dir = p.bundle
-	cmd.Stdin = p.stdio.stdin
-	cmd.Stdout = p.stdio.stdout
-	cmd.Stderr = p.stdio.stderr
-	// Call out to setPDeathSig to set SysProcAttr as elements are platform specific
-	cmd.SysProcAttr = setPDeathSig()
-
-	if err := cmd.Start(); err != nil {
-		if exErr, ok := err.(*exec.Error); ok {
-			if exErr.Err == exec.ErrNotFound || exErr.Err == os.ErrNotExist {
-				return fmt.Errorf("%s not installed on system", p.runtime)
-			}
-		}
-		return err
-	}
-	p.stdio.stdout.Close()
-	p.stdio.stderr.Close()
-	if err := cmd.Wait(); err != nil {
-		if _, ok := err.(*exec.ExitError); ok {
-			return errRuntime
-		}
-		return err
-	}
-	data, err := ioutil.ReadFile("pid")
-	if err != nil {
-		return err
-	}
-	pid, err := strconv.Atoi(string(data))
-	if err != nil {
-		return err
-	}
-	p.containerPid = pid
-	return nil
-}
-
 func (p *process) pid() int {
 	return p.containerPid
 }
 
 func (p *process) delete() error {
 	if !p.state.Exec {
-		out, err := exec.Command(p.runtime, append(p.state.RuntimeArgs, "delete", p.id)...).CombinedOutput()
+		out, err := exec.Command(p.runtime, append(p.state.RuntimeArgs, "stop", p.id)...).CombinedOutput()
```

Comment:
@mlaventure: sorry about this. I was doing some transient testing with an older version of the runtime (it was based on the oci spec that had runc start stop etc). This wasn't meant to be a part of the changeset. I've reverted it to delete.